carma_wm/src/TrafficControl.cpp

```diff
@@ -51,7 +51,22 @@ void fromBinMsg(const autoware_lanelet2_msgs::MapBin& msg, std::shared_ptr<carma
   std::stringstream ss;
   ss << data_str;
   boost::archive::binary_iarchive oa(ss);
+
   oa >> *gf_ptr;
+
+  if (!lanelet_map)
+    return;
```

Comment:
Are there two use cases for this function now? It looks like some of the logic gets run when there is no-map provided. Are both cases being used?

Reply:
First case where the map is not required is easy for unit test and the map_update_debugger.
Otherwise there is no solid reason not to enforce it at the start.

Reply:
This should be specified in the function comments

Reply:
Added to function comments